File: OldVBCode/MCTool+/MCToolPlus/MCTool+/SharedCode.vb
Imports System.IO
Module SharedCode
    Public Class ErrorHandler
        Public Shared Sub HandleError(GenError As Exception)
            Select Case GenError.GetType
                Case GetType(AccessViolationException)
                    MsgBox("Unable to get access to the specified directory.", MsgBoxStyle.Exclamation, "Access Error")
                Case GetType(UnauthorizedAccessException)
                    MsgBox("Unable to get write access to specified directory.", MsgBoxStyle.Exclamation, "Write Access Error")
                Case GetType(FileNotFoundException)
                    MsgBox("Unable to find the specified file.", MsgBoxStyle.Exclamation, "File not found Error")
                Case GetType(DirectoryNotFoundException)
                    MsgBox("Unable to find the specified directory.", MsgBoxStyle.Exclamation, "Directory not found Error")
                Case GetType(IOException)
                    MsgBox($"Unable to create/modify the specified file.", MsgBoxStyle.Exclamation, "I/O Error")
                Case Else
                    MsgBox("Unknown Error", MsgBoxStyle.Critical, "Application")
            End Select
        End Sub
    End Class
End Module

File: MCToolPlus/SharedCode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using MCToolPlus.Forms;
using System.IO;

namespace MCToolPlus
{
    static class SharedCode
    {

        public static void OpenForm(ref Form form,string ClassForm) //ref specifica che il void può cambiare il valore originale della variabile
        {

            if (form.IsDisposed)
            {
                switch (ClassForm)
                {
                    case "MainMenu":
                        form = new MainMenu();
                        break; //esce dallo switch
                    case "WorldManager":
                        form = new WorldManager();
                        break;
                    case "LogForm":
                        form = new LogForm();
                        break;
                    default:
                        throw new System.TypeLoadException();
                }
            }
            form.Show();
        }

        public static void HandleError(Exception ex)
        {
            if (ex.GetType() == typeof(AccessViolationException)) {MessageBox.Show("Unable to get access to the specified directory.", "Access Error",MessageBoxButtons.OK,MessageBoxIcon.Error);}
            else if (ex.GetType() == typeof(UnauthorizedAccessException)) { MessageBox.Show("Unable to get write access to specified directory.", "Write Access Error", MessageBoxButtons.OK, MessageBoxIcon.Error); }
            else if (ex.GetType() == typeof(FileNotFoundException)) { MessageBox.Show("Unable to find the specified file.", "File Not Found Error", MessageBoxButtons.OK, MessageBoxIcon.Error); }
            else if (ex.GetType() == typeof(DirectoryNotFoundException)) { MessageBox.Show("Unable to find the specified directory.", "Directory Not Found Error", MessageBoxButtons.OK, MessageBoxIcon.Error); }
            else if (ex.GetType() == typeof(IOException)) { MessageBox.Show("Unknown Error", "Application", MessageBoxButtons.OK, MessageBoxIcon.Error); }
        }
    }
}
